services/repository/mongodb/service.go

```diff
@@ -0,0 +1,173 @@
+// Returns the new service or error if something goes wrong
+func NewMongodbRepositoryService(configurationService configuration.ConfigurationContract) (repository.RepositoryContract, error) {
+
+	if configurationService == nil {
+		return nil, commonErrors.NewArgumentNilError("configurationService", "configurationService is required")
+	}
+
+	dbConnectionString, err := configurationService.GetDbConnectionString()
+	if err != nil {
+		return nil, err
+	}
+
+	return &mongodbRepositoryService{
+		dbConnectionString: dbConnectionString,
+	}, nil
+}
+
+// ReadTenant read an existing tenant
+// context: Optional The reference to the context
+// request: Mandatory. The request to read an existing tenant
+// Returns either the result of reading an exiting tenant or error if something goes wrong.
+func (service *mongodbRepositoryService) ReadTenant(
+	ctx context.Context,
+	request *repository.ReadTenantRequest) (*repository.ReadTenantResponse, error) {	
+
+		// Todo : move line 47-57 to a method
+		clientOptions := options.Client().ApplyURI(service.dbConnectionString)
+		client, err := mongo.Connect(context.TODO(), clientOptions)		
+		if err != nil {
+			return nil, repository.NewUnknownErrorWithError("Could not connect to mongodb database.", err)
+		}
+
+		//Todo : make sure if ping is necessary
+		err = client.Ping(context.TODO(), nil)
+		if err != nil {
+    		return nil, repository.NewUnknownErrorWithError("database is not accessible.", err)
+		}
+
+		//Todo : Do we need to have GetDbName in configuration service ? for now it is just hard coded
+		collection := client.Database("tenantdb").Collection("tenants")
+
+		filter := bson.D{{"TenantID", request.TenantID}}
+		var tenant models.Tenant
+
+		err = collection.FindOne(context.TODO(), filter).Decode(&tenant)
+		if err != nil {
+			return nil, repository.NewTenantNotFoundError(request.TenantID)
+		}
+
+		return &repository.ReadTenantResponse{Tenant: tenant}, nil
+	}
+
+
+// CreateTenant creates a new tenant.
+// context: Optional The reference to the context
+// request: Mandatory. The request to create a new tenant
+// Returns either the result of creating new tenant or error if something goes wrong.
+func (service *mongodbRepositoryService) CreateTenant(
+	ctx context.Context,
+	request *repository.CreateTenantRequest) (*repository.CreateTenantResponse, error) {
+
+	clientOptions := options.Client().ApplyURI(service.dbConnectionString)
+	client, err := mongo.Connect(context.TODO(), clientOptions)		
+	if err != nil {
+		return nil, repository.NewUnknownErrorWithError("Could not connect to mongodb database.", err)
+	}
+
+	//Todo : make sure if ping is necessary
+	err = client.Ping(context.TODO(), nil)
+	if err != nil {
+		return nil, repository.NewUnknownErrorWithError("database is not accessible.", err)
+	}
+
+ 	//Todo : Do we need to have GetDbName in configuration service ? for now it is just hard coded
+	collection := client.Database("tenantdb").Collection("tenants")
```

Comment:
name of database and collection should either come from the configuration service or must be defined as constant on top of this file